### PR TITLE
Configurable maxConcurrentReconciles and pollInterval with sensible defaults

### PR DESCRIPTION
**Pull request by turkenh**
### Description of your changes

This PR makes maxConcurrentReconciles and pollInterval configurable with the proposed defaults experimented [here](https://github.com/upbound/upjet/issues/116#issuecomment-1308941093).
It follows the same approach we used in the community providers to use input arguments as internal configurations.

Fixes https://github.com/upbound/upjet/issues/116

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested
 
See https://github.com/upbound/upjet/issues/116#issuecomment-1308941093

### CODE REVIEW

File: cmd/provider/main.go [hunk 1]
```diff
@@ -37,13 +37,14 @@ func main() {
 	var (
 		app                = kingpin.New(filepath.Base(os.Args[0]), "AWS support for Crossplane.").DefaultEnvars()
 		debug              = app.Flag("debug", "Run with debug logging.").Short('d').Bool()
-		syncPeriod         = app.Flag("sync", "Controller manager sync period such as 300ms, 1.5h, or 2h45m").Short('s').Default("1h").Duration()
+		syncInterval       = app.Flag("sync", "Sync interval controls how often all resources will be double checked for drift.").Short('s').Default("1h").Duration()
+		pollInterval       = app.Flag("poll", "Poll interval controls how often an individual resource should be checked for drift.").Default("10m").Duration()
 		leaderElection     = app.Flag("leader-election", "Use leader election for the controller manager.").Short('l').Default("false").OverrideDefaultFromEnvar("LEADER_ELECTION").Bool()
+		maxReconcileRate   = app.Flag("max-reconcile-rate", "The global maximum rate per second at which resources may checked for drift from the desired state.").Default("10").Int()
```
Comment (ulucinar) [inline cmd/provider/main.go:43]:
```suggestion
		maxReconcileRate   = app.Flag("max-reconcile-rate", "The global maximum rate per second at which resources may be checked for drift from the desired state.").Default("10").Int()
```

File: cmd/provider/main.go [hunk 2]
```diff
@@ -37,13 +37,14 @@ func main() {
 	var (
 		app                = kingpin.New(filepath.Base(os.Args[0]), "AWS support for Crossplane.").DefaultEnvars()
 		debug              = app.Flag("debug", "Run with debug logging.").Short('d').Bool()
-		syncPeriod         = app.Flag("sync", "Controller manager sync period such as 300ms, 1.5h, or 2h45m").Short('s').Default("1h").Duration()
+		syncInterval       = app.Flag("sync", "Sync interval controls how often all resources will be double checked for drift.").Short('s').Default("1h").Duration()
+		pollInterval       = app.Flag("poll", "Poll interval controls how often an individual resource should be checked for drift.").Default("10m").Duration()
```
Comment (ulucinar) [inline cmd/provider/main.go:41]: Why are we increasing the poll interval from 1m to 10m?

File: cmd/provider/main.go [hunk 3]
```diff
@@ -91,8 +93,8 @@ func main() {
 		Options: xpcontroller.Options{
 			Logger:                  log,
 			GlobalRateLimiter:       ratelimiter.NewGlobal(*maxReconcileRate),
-			PollInterval:            1 * time.Minute,
-			MaxConcurrentReconciles: 1,
+			PollInterval:            *pollInterval,
+			MaxConcurrentReconciles: *maxReconcileRate,
```
Comment (ulucinar) [inline cmd/provider/main.go:97]: Should we have separate parameters for the max reconcile rate (which is a global rate shared between all reconcilers) and the max concurrent reconciles (which is a count of workers consuming the workqueue of a controller)?

Comment (turkenh) [reply cmd/provider/main.go:97]: @ulucinar I would like to be consistent with other providers and @negz has a detailed description [here](https://github.com/crossplane/crossplane/issues/2595) on why it makes sense to use the same parameter for both.

File: cmd/provider/main.go (hunk 2 above)
Comment (turkenh) [reply cmd/provider/main.go:41]: This is the proposed value after my experiments in https://github.com/upbound/upjet/issues/116#issuecomment-1308941093 with the motivation below:

In upjet based controllers a typical reconcile loop tooks longer than a native provider due to underlying implementation. If we go with a polling interval like `1m`, we immediately cause the work queue to fill up especially when there are multiple instances of the same kind. I believe it is more reasonable to default to a higher value if a typical reconcile loop takes is in the order of ~10s (i.e. refresh/plan).

File: cmd/provider/main.go (hunk 3 above)
Comment (ulucinar) [reply cmd/provider/main.go:97]: Thanks for the pointer @turkenh. I agree, we had better follow the convention suggested there.

**Review by ulucinar — APPROVED** [review]
Thanks @turkenh, lgtm.

**Pull request merged**